client/containers/Budget/index.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component, PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+import * as AppActions from 'actions';
+import BudgetGrid from 'components/BudgetGrid';
+
+class Budget extends Component {
+  static propTypes = {
+    transactions: PropTypes.array.isRequired,
+    categories: PropTypes.object.isRequired
+  }
+
+  render() {
+    const { transactions, categories } = this.props;
+    const data = { transactions, categories };
+
+    return (
+      <div>
+        <h1>Budget</h1>
+        <BudgetGrid data={data} />
+      </div>
+    );
+  }
+}
+
+function mapStateToProps(state) {
+  const { transactions } = state;
+
+  return {
+    transactions: transactions.transactions,
+    categories: transactions.categories
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    actions: bindActionCreators(AppActions, dispatch)
+  };
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(Budget);
```

Comment:
can we please use `@connect` decorator instead?

```jsx
@connect(
  ({ transactions }) => ({
    transactions: transactions.transactions,
    categories: transactions.categories,
  }),
  {
    foo: AppActions.foo,
  }
class Budget extends Component {
  static propTypes = {
    transactions: PropTypes.array.isRequired,
    categories: PropTypes.object.isRequired
  }

  render() {
    const { transactions, categories } = this.props;
    const data = { transactions, categories };

    return (
      <div>
        <h1>Budget</h1>
        <BudgetGrid data={data} />
      </div>
    );
  }
}

export default Budget;
```

I definitely recomment specifying every action inside @connect instead of spreading or using object because you'll later know where your functions (props) are coming from